### PR TITLE
Required forms

### PR DESCRIPTION
**Pull request by kwm14**
## Done
- Design discussion can be found here: https://github.com/ubuntudesign/vanilla-design/issues/34
- Currently we don't have a style for required form fields
- Created a 'required' class style which can be added to form labels, was best to build a demo proposal to see it in situ against current implementation (MAAS predominantly)

## QA

- Pull code
- Run `./run serve --watch`
- Open http://0.0.0.0:8101/vanilla-framework/examples/templates/required-forms/
- Look at proposed styling vs current styling and review

## Details
Demo proposal raised from this issue https://github.com/ubuntudesign/vanilla-design/issues/34

### CODE REVIEW

**Comment by anthonydillon** [comment]
@kwm14 looks good. A few points:
- the class should be `is-required` so it's inline with our other state classes.
- the example should be simplified and placed inside the forms directory. `/forms/forms-required.html`.
- There is no additional documentation added with this branch. The should be a required input selection on the forms docs

I know this is a demo but with the points above this branch would be in a mergable state once at l agreed on.

**Comment by kwm14** [comment]
@anthonydillon thanks for your feedback 👍 

I've updated the class name 😄 

> The example should be simplified and placed inside the forms directory. `/forms/forms-required.html`. 

Do I need to create a new directory? As I can't find the one you've mentioned inside the project

> There is no additional documentation added with this branch. The should be a required input selection on the forms docs

Ok. I'll add a section to docs to explain when this style can be applied to forms 👍

**Comment by spencerbygraves** [comment]
@kwm14 I think it looks fine visually, if this is agreed from a UX point of view.

**Comment by kwm14** [comment]
Cheers @anthonydillon 👍 

@spencerbygraves thanks for your review, could you update labels too 😄 

@matthewpaulthomas ok from UX?

**Comment by matthewpaulthomas** [comment]
I can’t tell whether this example page is public anywhere. If it is, so that authors may copy and paste its code, please add the `required` attribute (that is, `<input required class=…`) for accessibility.

Apart from that, this looks fine to me.

**Comment by kwm14** [comment]
Thanks @matthewpaulthomas this PR is for showcase a demo for UI/UX sign-off ✅ Good spot on the attribute for accessibility I will make sure I update 👍 

I'm going create a new branch for Code and QA and will add the example [here](https://github.com/vanilla-framework/vanilla-framework/tree/develop/examples/patterns/forms.) as requested by @anthonydillon so that it's documented.

**Comment by kwm14** [comment]
@anthonydillon added example and documented, signed off by design so just need Code and QA review 👍

File: scss/_base_forms.scss [hunk 1]
```diff
@@ -146,6 +146,13 @@ $input-margin-bottom: $spv-inter--scaleable - $spv-nudge * 2;
     display: block;
     margin-bottom: $spv-inter--condensed + $spv-nudge-compensation;
 
+    &.is-required::after {
+      color: $color-negative;
+      content: '*';
+      left: 0.25rem;
```
Comment (anthonydillon) [inline scss/_base_forms.scss]: Replace `0.25rem` with `$spv-intra--condensed`

File: examples/patterns/forms/forms-required.html [hunk 2]
```diff
@@ -0,0 +1,15 @@
+---
+layout: default
+title: Forms / Required
+category: _patterns
+---
+
+      <form>
```
Comment (anthonydillon) [inline examples/patterns/forms/forms-required.html]: Why is this example indented?

File: examples/patterns/forms/forms-required.html [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+        <div class="p-form-validation is-error">
+          <label for="exampleTextInputError" required class="is-required">Email address</label>
+          <input class="p-form-validation__input" type="text" id="exampleTextInputError" placeholder="e.g joe@bloggs.com">
```
Comment (anthonydillon) [inline examples/patterns/forms/forms-required.html]: Required should be in the input not the label. It's fine to leave the class on the label.

File: scss/_base_forms.scss (hunk 1 above)
Comment (kwm14) [reply scss/_base_forms.scss]: Will update. Thanks

File: examples/patterns/forms/forms-required.html (hunk 2 above)
Comment (kwm14) [reply examples/patterns/forms/forms-required.html]: I blame Atom Beautify plugin 😉

File: examples/patterns/forms/forms-required.html (hunk 3 above)
Comment (kwm14) [reply examples/patterns/forms/forms-required.html]: Sure thing 👍

**Comment by kwm14** [comment]
@anthonydillon updates done, ready for another review 👀

**Pull request merged**